### PR TITLE
style: One-time code clean-up using Sourcery

### PR DESCRIPTION
**Pull request by BoboTiG**
### What I did

Run Sourcery on the whole source code + adapted changes here and there + minor CI adjustments.
I know it might be a big change, and maybe such PRs are not welcome. I would be happy to make adjustments, or simply close the PR.

Positive points:
- it follows the style guide (like using f-strings everywhere)
- the CI is a little bit more unifyied (black arguments are the same whenever you run it from tox, python, or the GitHub workflow)
- removed a lot of code (on a personal opinion I find it cleaner)

I wanted to propose such changes before doing anything else :)

### Commit message

```
style: One-time code clean-up using Sourcery
```

### Description for the changelog

One-time code clean-up using Sourcery.

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: tests/grammar/test_grammar.py [hunk 1]
```diff
@@ -39,11 +39,10 @@ def test_basic_grammar_empty():
 
 def utf8_encodable(terminal: str) -> bool:
     try:
-        if "\x00" not in terminal and "\\ " not in terminal and "\x0c" not in terminal:
-            terminal.encode("utf-8-sig")
-            return True
-        else:
+        if "\x00" in terminal or "\\ " in terminal or "\x0c" in terminal:
             return False
+        terminal.encode("utf-8-sig")
```
Comment (fubuloubu) [inline tests/grammar/test_grammar.py:44]: Probably can tighten the try catch to just this line

File: vyper/compiler/utils.py [hunk 2]
```diff
@@ -4,12 +4,11 @@
 
 
 def build_gas_estimates(func_ts: Dict[str, ContractFunctionT]) -> dict:
-    # note: `.gas_estimate` is added to ContractFunctionT._ir_info
-    # in vyper/semantics/types/function.py
```
Comment (fubuloubu) [inline vyper/compiler/utils.py:8]: Probably keep this

**Review by fubuloubu — COMMENTED** [review]
Great tool, thanks!

File: vyper/compiler/utils.py (hunk 2 above)
Comment (BoboTiG) [reply vyper/compiler/utils.py:8]: Yup, it's kept as the function docstring rather ;)

Comment (fubuloubu) [reply vyper/compiler/utils.py:8]: Oh I mean keep it as it was, not quite a docstring, just a dev comment

**Comment by BoboTiG** [comment]
era-compiler-tester is failing, but it's the same error as on the main branch. I guess it's not related to my changes.

**Comment by charles-cooper** [comment]
i think if we want to use this tool, we should add it as a CI integration?

**Comment by BoboTiG** [comment]
It would be a good idea. To do so, someone from the team with enough GitHub rights should do it via https://sourcery.ai/. No configuration to store into the repository.

**Comment by charles-cooper** [comment]
i can't say i'm a huge fan of handing over formatting of our source code to some proprietary (and probably unstable?) tool. and i do like a lot of the changes stylistically but not all of them, like using ternary operators instead of if- statements here https://github.com/vyperlang/vyper/pull/3495/files#diff-a8a3d37294d14676d136ab2a4f04c5112e971b2cfad21cfd3f77de75b1273b62R635

**Comment by BoboTiG** [comment]
That's why I propose a one-time "clean-up" patch. I use such a tool on a daily basis for all perosnal + professionnal projects without any issue for years (it is stable). But I get it, it's not FOSS, and using it, or not, on all PRs is up to the team. I put such tools in the same kind as `black`, where it allowes to have non-personal code improvements (some of them could be achieved via `flake8-simplify` and others, but they are not at the same level of optimizations).

You have enough rights to edit the PR I guess, please yourself with adjustments. As I said in the description, the PR can be closed, and we move on :)

**Comment by charles-cooper** [comment]
closing as i don't really have time to go through and verify all the changes + figure out which changes to revert / which ones to keep

**Pull request closed without merging**